Apps/Src/UpdateVelocity.c

```diff
@@ -48,23 +53,61 @@ void Task_UpdateVelocity(void *p_arg)
         uint8_t accelPedalPercent = Pedals_Read(ACCELERATOR);
         uint8_t brakePedalPercent = Pedals_Read(BRAKE);
 
+        //Edge detector to toggle the cruise enable on only the rising edge
+        State currEnable = Switches_Read(CRUZ_EN);
+        if((prevEnable == OFF) && (currEnable == ON)){ //we went from off to on, so we should toggle it
+            prevEnable = ON;
+            cruiseEnable ^= 1; 
+        } else if ((currEnable == OFF)){ //went from on to OFF, so reset the prev state
+            prevEnable = OFF;
+        }
+
+        //Edge detector to toggle the cruise set on only the rising edge
+
+        if(cruiseEnable){
+            State currSet = Switches_Read(CRUZ_ST);
+            if((prevSet == OFF) && (currSet == ON)){
+                prevSet = ON;
+                cruiseSet ^= 1;
+            } else if ((currSet == OFF)){
+                prevSet = OFF;
+            }
+        } else {
+            cruiseSet = OFF;
+        }
```

Comment:
I think this could be more readable with something like
```C
// declare currEnabled before the start of the while loop
// declare these variables as bools

// edge detector
prevEnabled = currEnabled;
currEnabled = Switches_Read(CRUZ_EN);
if (!prevEnabled && currEnabled) {
    cruiseSet = !cruiseSet;

    if  (cruiseSet) {
        // turn on cruise control
    } else {
        // turn off cruise control
    }
}
```